frontend/src/scenes/cohorts/CohortEdit.tsx

```diff
@@ -121,6 +122,25 @@ export function CohortEdit({ id }: CohortLogicProps): JSX.Element {
         return <NotFound object="cohort" />
     }
 
+    if (cohort.deleted) {
+        return (
+            <div>
+                <LemonBanner type="error">The cohort '{cohort.name}' has been soft deleted.</LemonBanner>
+                <ScenePanel>
+                    <ButtonPrimitive
+                        disabled={cohortLoading}
+                        onClick={() => {
+                            restoreCohort()
+                        }}
+                        menuItem
+                    >
+                        <IconRefresh /> Restore
+                    </ButtonPrimitive>
+                </ScenePanel>
+            </div>
+        )
+    }
```

Comment:
@adamleithp @mariusandra Should we introduce a standard "Restore" banner pattern like this, or instead enforce `<NotFound ...` here?